staking/types/validator.go

```diff
@@ -259,6 +262,14 @@ func (w *ValidatorWrapper) SanityCheck() error {
 		)
 	}
 
+	allKeys := map[shard.BlsPublicKey]struct{}{}
```

Comment:
we can never add a duplicate key to `validator.SlotPubKeys`, so why check for duplicates?

Reply:
I see this logic is processing the input. Then it is fine I guess.